src/mesh.cpp

```diff
@@ -44,12 +44,13 @@ void applyFacesShading(video::SColor &color, const v3f &normal)
 {
 	// Many special drawtypes have normals set to 0,0,0 and this
 	// must result in maximum brightness (no face shadng).
-	if (normal.Y < -0.5f)
-		applyShadeFactor (color, 0.447213f);
-	else if (normal.X > 0.5f || normal.X < -0.5f)
-		applyShadeFactor (color, 0.670820f);
-	else if (normal.Z > 0.5f || normal.Z < -0.5f)
-		applyShadeFactor (color, 0.836660f);
+	float x2 = normal.X * normal.X;
+	float y2 = normal.Y * normal.Y;
+	float z2 = normal.Z * normal.Z;
+	if (normal.Y < 0)
```

Comment:
Since you are using ```x2 > 1e-3``` below to do a 'not zero' check with a small safety margin how about using that safety margin here too? Otherwise may as well use ```else if ((x2 > 0) || (z2 > 0))``` below.